alacritty/src/input/keyboard.rs

```diff
@@ -99,6 +102,33 @@ impl<T: EventListener, A: ActionContext<T>> Processor<T, A> {
         }
     }
 
+    fn alt_send_esc(&mut self, key: &KeyEvent, text: &str) -> bool {
+        #[cfg(not(target_os = "macos"))]
+        let alt_send_esc = self.ctx.modifiers().state().alt_key();
+
+        #[cfg(target_os = "macos")]
+        let alt_send_esc = {
+            let option_as_alt = self.ctx.config().window.option_as_alt();
+            self.ctx.modifiers().state().alt_key()
+                && (option_as_alt == OptionAsAlt::Both
+                    || (option_as_alt == OptionAsAlt::OnlyLeft
+                        && self.ctx.modifiers().lalt_state() == ModifiersKeyState::Pressed)
+                    || (option_as_alt == OptionAsAlt::OnlyRight
+                        && self.ctx.modifiers().ralt_state() == ModifiersKeyState::Pressed))
+        };
+
+        match key.logical_key {
+            Key::Named(named) => {
+                if named.to_text().is_some() {
+                    alt_send_esc
+                } else {
+                    self.ctx.modifiers().state().alt_key()
+                }
```

Comment:
What is this branch for? I'd assume it's for keys that have no characters attached, so is it specifically to handle modifier presses themselves? Alt still shouldn't send esc if that's the case on macOS, right?

Reply:
That's for `Arrow`/`PageUp` keys where `Alt` should actually be `Alt` and not `Option` because they have alternate character we're trying to avoid with all of that mess.